cluster-autoscaler/cloudprovider/clusterapi/internal/clusterapi_manager.go

```diff
@@ -0,0 +1,111 @@
+//go:generate ./vendor-cluster-api.sh
+
+package internal
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider/clusterapi/types"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	v1alpha1apis "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+)
+
+const (
+	refreshInterval = 30 * time.Second
+)
+
+type clusterManager struct {
+	lastRefresh          time.Time
+	clusterSnapshot      *clusterSnapshot
+	clusterSnapshotMutex sync.Mutex
```

Comment:
by convention we place locks directly above the fields they protect and add a comment - this should be:

```
// protects the cluster snapshot
clusterSnapshotMutex sync.Mutex
clusterSnapshot *clusterSnapshot

... other fields
```

Reply:
Done.